app/controllers/tools_controller.rb

```diff
@@ -1,60 +1,129 @@
 class ToolsController < ApplicationController
   SIMILAR_ENTITIES_PER_PAGE = 75
 
+  class MergeModes
+    SEARCH  = 'search'.freeze
+    EXECUTE = 'execute'.freeze
+    REQUEST = 'request'.freeze
+    REVIEW  = 'review'.freeze
+    ALL = [SEARCH, EXECUTE, REQUEST, REVIEW].freeze
+  end
+
   before_action :authenticate_user!
   before_action :set_entity, only: [:bulk_relationships]
-  before_action -> { check_permission 'merger' }, only: [:merge_entities, :merge_entities!]
-  before_action :parse_merge_params, only: [:merge_entities]
-  before_action :set_source_and_dest, only: [:merge_entities!]
+  before_action :admins_only, only: [:merge_entities!]
+  before_action :parse_merge_mode, only: [:merge_entities, :merge_entities!]
+  before_action :parse_merge_get_params, only: [:merge_entities]
+  before_action :parse_merge_post_params, only: [:merge_entities!]
+  before_action :parse_rendundant_review_params, only: [:redundant_merge_review]
 
-  def bulk_relationships
-  end
+  def bulk_relationships; end
 
   # GET /tools/merge
-  # There are 3 combinatations of params this accepts:
-  # - source
-  # - source and query
-  # - source and dest
-  def merge_entities
-    set_similar_entities if @merge_mode == :search
-  end
+  # possible params: mode, source, dest, query
+  def merge_entities; end
 
   # POST /tools/merge
-  # do the merge
+  # do the merge.freeze
   def merge_entities!
-    EntityMerger.new(source: @source, dest: @dest).merge!
-    redirect_to @dest
+    case @merge_mode
+    when MergeModes::EXECUTE
+      @source.merge_with(@dest)
+      redirect_to @dest
+    when MergeModes::REVIEW
+      @merge_request.send("#{@decision}_by!".to_sym, current_user)
+      redirect_to @merge_request.source
+    end
   end
 
+  # GET /tools/redundant
```

Comment:
perhaps the path could be: `  /tools/merge/redundant  ` since /redundant isn't the name of a tool?

Reply:
whoops! that *is* what the path is. this is a typo.

Reply:
oh. haha!